server/channels/api4/channel_test.go

```diff
@@ -2806,28 +2889,37 @@ func TestDeleteChannel(t *testing.T) {
 		_, err = client.DeleteChannel(context.Background(), sdPrivateChannel.Id)
 		require.NoError(t, err)
 	})
-	th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
-		th.LoginBasic()
-		publicChannel5 := th.CreatePublicChannel()
-		c.Logout(context.Background())
 
-		c.Login(context.Background(), user.Id, user.Password)
-		resp, err := c.DeleteChannel(context.Background(), publicChannel5.Id)
-		require.Error(t, err)
-		CheckUnauthorizedStatus(t, resp)
+	th.LoginBasic()
+	publicChannel5 := th.CreatePublicChannel()
+	_, err := c.Logout(context.Background())
+	require.NoError(t, err)
 
-		resp, err = c.DeleteChannel(context.Background(), "junk")
-		require.Error(t, err)
-		CheckUnauthorizedStatus(t, resp)
+	// Other users can't delete the channel
+	_, _, err = c.Login(context.Background(), user2.Email, user2.Password)
+	require.NoError(t, err)
+	resp, err := c.DeleteChannel(context.Background(), publicChannel5.Id)
+	require.Error(t, err)
+	CheckForbiddenStatus(t, resp)
 
-		c.Logout(context.Background())
-		resp, err = c.DeleteChannel(context.Background(), GenerateTestID())
-		require.Error(t, err)
-		CheckUnauthorizedStatus(t, resp)
+	resp, err = c.DeleteChannel(context.Background(), "junk")
+	require.Error(t, err)
+	CheckBadRequestStatus(t, resp)
 
-		_, err = client.DeleteChannel(context.Background(), publicChannel5.Id)
-		require.NoError(t, err)
-	})
+	_, err = c.Logout(context.Background())
+	require.NoError(t, err)
+	resp, err = c.DeleteChannel(context.Background(), GenerateTestID())
+	require.Error(t, err)
+	CheckUnauthorizedStatus(t, resp)
+
+	// The creator can delete the channel
+	_, err = c.Logout(context.Background())
+	require.NoError(t, err)
+	_, _, err = c.Login(context.Background(), user.Email, user.Password)
+	require.NoError(t, err)
+	resp, err = c.DeleteChannel(context.Background(), publicChannel5.Id)
+	require.NoError(t, err)
+	CheckOKStatus(t, resp)
```

Comment:
The test was just broken. The `c.Login` was wrong, leading to wrongly assumed status codes.